packages/realm_dart/lib/src/realm_object.dart

```diff
@@ -719,12 +728,24 @@ extension RealmObjectChangesInternal<T extends RealmObject> on RealmObjectChange
 class RealmObjectNotificationsController<T extends RealmObjectBase> extends NotificationsController {
   T realmObject;
   late final StreamController<RealmObjectChanges<T>> streamController;
+  List<String>? keyPaths;
+  int? classKey;
 
-  RealmObjectNotificationsController(this.realmObject);
+  RealmObjectNotificationsController(this.realmObject, [List<String>? keyPaths]) {
+    if (keyPaths != null) {
+      this.classKey = realmObject.realm.metadata.getByType(T).classKey;
+      this.keyPaths = keyPaths;
+
+      if (keyPaths.any((element) => element.isEmpty)) {
+        throw RealmException("It is not allowed to have empty key paths.");
+      }
+      realmCore.verifyKeyPath(realmObject, keyPaths, classKey);
+    }
+  }
 
   @override
   RealmNotificationTokenHandle subscribe() {
-    return realmCore.subscribeObjectNotifications(realmObject, this);
+    return realmCore.subscribeObjectNotifications(realmObject, this, keyPaths, classKey);
```

Comment:
I notice you pass in both the `realmObject` and the `classKey` where the later can be inferred from the former. I suppose this is to avoid the overhead of calling `realmObject.realm.metadata.getByType(T).classKey;` twice. 

Maybe that is a bit of premature optimization. In particular since we eventually want to avoid creating `keypathsNative` twice anyway.